benchmark/syclblas/blas3/symm.cpp

```diff
@@ -0,0 +1,163 @@
+         bool* success) {
+  const index_t k = side == 'l' ? m : n;
+
+  index_t lda = k;
+  index_t ldb = m;
+  index_t ldc = m;
+
+  blas::SB_Handle& sb_handle = *sb_handle_ptr;
+
+  // Matrices
+  std::vector<scalar_t> a = blas_benchmark::utils::random_data<scalar_t>(k * k);
+  std::vector<scalar_t> b = blas_benchmark::utils::random_data<scalar_t>(m * n);
+  std::vector<scalar_t> c =
+      blas_benchmark::utils::const_data<scalar_t>(m * n, 0);
+
+  auto a_gpu = blas::make_sycl_iterator_buffer<scalar_t>(a, k * k);
+  auto b_gpu = blas::make_sycl_iterator_buffer<scalar_t>(b, m * n);
+  auto c_gpu = blas::make_sycl_iterator_buffer<scalar_t>(c, m * n);
+
+#ifdef BLAS_VERIFY_BENCHMARK
+  // Run a first time with a verification of the results
+  std::vector<scalar_t> c_ref = c;
+  const char side_str[2] = {side, '\0'};
+  const char uplo_str[2] = {uplo, '\0'};
+  reference_blas::symm(side_str, uplo_str, m, n, alpha, a.data(), lda, b.data(), ldb,
+                       beta, c_ref.data(), ldc);
+  std::vector<scalar_t> c_temp = c;
+  {
+    auto c_temp_gpu = blas::make_sycl_iterator_buffer<scalar_t>(c_temp, m * n);
+    auto event = _symm(sb_handle, side, uplo, m, n, alpha, a_gpu, lda, b_gpu,
+                       ldb, beta, c_temp_gpu, ldc);
+    sb_handle.wait(event);
+  }
+
+  std::ostringstream err_stream;
+  if (!utils::compare_vectors(c_temp, c_ref, err_stream, "")) {
+    const std::string& err_str = err_stream.str();
+    state.SkipWithError(err_str.c_str());
+    *success = false;
+  };
+#endif
+
+  auto blas_method_def = [&]() -> std::vector<cl::sycl::event> {
+    auto event = _symm(sb_handle, side, uplo, m, n, alpha, a_gpu, lda, b_gpu,
+                       ldb, beta, c_gpu, ldc);
+    sb_handle.wait(event);
+    return event;
+  };
+
+  // Warmup
+  blas_benchmark::utils::warmup(blas_method_def);
+  sb_handle.wait();
+
+  blas_benchmark::utils::init_counters(state);
+
+  // Measure
+  for (auto _ : state) {
+    // Run
+    std::tuple<double, double> times =
+        blas_benchmark::utils::timef(blas_method_def);
+
+    // Report
+    blas_benchmark::utils::update_counters(state, times);
+  }
+
+  {
+    // The counters are double. We convert m, n and k to double to avoid
+    // integer overflows for n_fl_ops and bytes_processed
+    double m_d = static_cast<double>(m);
+    double n_d = static_cast<double>(n);
+    double k_d = static_cast<double>(k);
+
+    state.counters["m"] = m_d;
+    state.counters["k"] = k_d;
+    state.counters["n"] = n_d;
+
+    double mem_readA = k_d * (k_d + 1) / 2;
+    double mem_readB = m_d * n_d;
+    double mem_writeC = m_d * n_d;
+    double mem_readC = (beta != scalar_t{0}) ? m_d * n_d : 0;
+    double total_mem =
+        (mem_readA + mem_readB + mem_readC + mem_writeC) * sizeof(scalar_t);
+    state.counters["bytes_processed"] = total_mem;
+    state.SetBytesProcessed(state.iterations() * total_mem);
+
+    double nflops_AtimesB = (2 * k_d - 1) * m_d * n_d;
```

Comment:
Could you double check these calculations based on the file that I have shared with you?